source/nodejs/adaptivecards-site/service-worker.js

```diff
@@ -0,0 +1,109 @@
+// Used pwa-builder as reference: https://github.com/pwa-builder/pwabuilder-serviceworkers/blob/master/serviceWorker5/pwabuilder-sw.js
+importScripts(
```

Comment:
@aarongustafson I registered the service worker and manifest within the designer html file. For the service worker, I set the defined scope to the designer directory at registration, but have it stored a directory up so that it can access the `content` folder. Do you think this is the correct approach?

Reply:
I’m not super familiar with Hexo and how this will all come out in the build, but I think that sounds right. Ultimately you want it to be in the directory you want it to govern. If it need to be able to operate on all fines in the site, the root is the right place for it. If you only want it to affect the Designer, it might make more sense to put it there so it doesn’t interfere with what’s happening outside of that directory.

Honestly, there’s probably a lot of good the SW can do for the site as a whole, so the root is probably the right place, even if you don’t add all those rules right now.

Reply:
@licanhua I add a minified version of the service worker